test/WasmTasksTests/WasmTask_Tests.cs

```diff
@@ -160,42 +161,220 @@ private static void ExecuteCommand(string command)
-            BuildEngine = new MockEngine();
+            public TemplateWasmTask() : base()
+            {
+                WasmFilePath = WasmTask_Tests.s_templateFilePath;
+                BuildEngine = new MockEngine();
+            }
         }
-    }
 
-    public class ConcatWasmTask : WasmTask
-    {
-        public ITaskItem? InputFile1 { get; set; }
-        public ITaskItem? InputFile2 { get; set; }
-        [Output]
-        public ITaskItem? OutputFile { get; set; }
+        public class ConcatWasmTask : WasmTask
+        {
+            public ITaskItem? InputFile1 { get; set; }
+            public ITaskItem? InputFile2 { get; set; }
+            [Output]
+            public ITaskItem? OutputFile { get; set; }
 
-        public ConcatWasmTask() : base()
+            public ConcatWasmTask() : base()
+            {
+                WasmFilePath = WasmTask_Tests.s_concatFilePath;
+                BuildEngine = new MockEngine();
+            }
+        }
+
+        public class DirectoryMergeWasmTask : WasmTask
         {
-            WasmFilePath = WasmTask_Tests.s_concatFilePath;
-            BuildEngine = new MockEngine();
+            public ITaskItem[]? Dirs { get; set; }
+            public string? MergedName { get; set; }
+            [Output]
+            public ITaskItem? MergedDir { get; set; }
+
+            public DirectoryMergeWasmTask() : base()
+            {
+                WasmFilePath = WasmTask_Tests.s_mergeFilePath;
+                BuildEngine = new MockEngine();
+            }
         }
-    }
+        private class CustomRustE2ETest
+        {
+            private readonly string _name;
+            private readonly string _templatePath;
+            private readonly string _testProjectPath;
 
-    public class DirectoryMergeWasmTask : WasmTask
-    {
-        public ITaskItem[]? Dirs { get; set; }
-        public string? MergedName { get; set; }
-        [Output]
-        public ITaskItem? MergedDir { get; set; }
+            public CustomRustE2ETest(string name, string toplevel, string executeRustCode, string getTaskInfoRustCode)
+            {
+                _name = name;
+                _templatePath = Path.Combine(SOLUTION_ROOT_PATH, s_paths[0]);
+                _testProjectPath = Path.Combine(TEST_PROJECT_PATH, "GeneratedRustTests", _name);
+
+                CopyTemplate();
+                CreateLibRs(toplevel, executeRustCode, getTaskInfoRustCode);
+                Compile();
+            }
+
+            private void CopyTemplate()
+            {
+                if (Directory.Exists(_testProjectPath))
+                {
+                    Directory.Delete(_testProjectPath, true);
+                }
+                DirectoryCopy(_templatePath, _testProjectPath, true);
+            }
+
+            private void CreateLibRs(string topLevel, string execute, string getTaskInfo)
+            {
+                string contents = $@"
+mod msbuild;
```

Comment:
Isn't it better to move it to some external assets and read the content from it?